Device handler for IKEA of Sweden STARKVIND Air purifier

arnemauer opened the pull request:
Quirk for the IKEA Starkvind Air purifier! #1215

**Supports:**
- All fan modes: Off, Auto, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5 & 5
- Air quality PM 2.5
- Device run time
- Filter run time (of the current placed filter)
- Filter status (need to replace filter) corresponds with the replace filter light on the Starkvind
- Child Lock (disables all controles)  corresponds with the child lock light on the Starkvind
- Disable Leds (disables all leds on the Starkvind)
- Filter lifetime (you can set the lifetime of a filter, default is 6 months = 259200 minuts). If filter run time exceeds this value, Filter status is true (and you need to replace the filter)

**This quirk also needs some adjustments in the ZHA code! 
See this pull request: https://github.com/home-assistant/core/pull/73450**

<img src="https://user-images.githubusercontent.com/268051/173425130-e92487ef-61b8-434b-b6e3-36e42c259252.png" width="400"><img src="https://user-images.githubusercontent.com/268051/173425175-17a5b9f3-fb84-4465-b48e-90e24c6bc2a4.png" width="200">

MattWestb commented:
I think the default reporting rate is little over kill for the air purifier and PM25 cluster that can spamming the network if the sensors is updating oft but it must being fixed in the ZHA core and cant being made in the quirk.

### GREAT WORK DONE !!!

@dmulcahey do you knowing where the default reporting is configured in ZHA core so the device is not spamming the network for the 3 new cluster ?

dmulcahey commented:
Each channel (basically a handler for a cluster) defines the report configs per attribute that it cares about. For example: https://github.com/home-assistant/core/blob/c660fae8d8dfd190186fc404ca77cc1bc4c0cda3/homeassistant/components/zha/core/channels/general.py#L49

arnemauer commented:
@MattWestb @dmulcahey Hmm, yeah it will send every minute a attribute report. Maybe change it to every 5 minuts or more? What is an usable interval?

See this pull request for a IkeaAirPurifier Cluster:
https://github.com/home-assistant/core/blob/12735b874e4d57c8f816cb8c7991d92e066f90e1/homeassistant/components/zha/core/channels/manufacturerspecific.py#L122

MattWestb commented:
I think 1 in the minute is OK then its main powered.
I was thinking the runtime counter was updated every second but then its every minute.
Also the PM25 is "filtered" so not sending so oft (Vindriktning is pulling the sensor 7 times every 20 seconds so raw data is 21 / min if not being filtered) so its also OK and getting nice graph in HA GUI :-))).

Then letting it and getting it merged !!

arnemauer commented:
* Added a test for de ikea air purifier.
* Instead of using the fan cluster, i've created a Ikea Air Purifier channel/cluster
* Updated sensors to work with the new channel/cluster
* Also updated the quirk to work with this changes :)

This quirk needs to be merged first before merging (https://github.com/home-assistant/core/pull/73450) to pass the new test.
